pinot-common/src/main/java/org/apache/pinot/common/Utils.java

```diff
@@ -145,4 +149,36 @@ public static void logVersions() {
 
     return componentVersions;
   }
+
+  public static String getGitInfo() {
+    // unless Utils was somehow loaded on the bootclasspath, this will not be null
+    // and will find all manifests
+    ClassLoader classLoader = Utils.class.getClassLoader();
+    if (classLoader != null) {
+      try {
+        Enumeration<URL> manifests = classLoader.getResources("META-INF/MANIFEST.MF");
+        while (manifests.hasMoreElements()) {
+          URL url = manifests.nextElement();
+          try (InputStream stream = url.openStream()) {
+            Manifest manifest = new Manifest(stream);
+            Attributes attributes = manifest.getMainAttributes();
+            if (attributes != null) {
+              String implementationTitle = attributes.getValue(Attributes.Name.IMPLEMENTATION_TITLE);
```

Comment:
nit: leave a comment that those info are injected via configs in that pom.xml file